docs/source/core_developers_guide/environment_setup.rst

```diff
@@ -144,3 +174,182 @@ install_sphinx
+
+
+Step Three: Start VM
+====================
+
+In order to start the vagrant VM, run:
+
+.. code-block:: console
+
+  % cd sawtooth-core/tools
+  % vagrant up
+
+.. note::
+
+   We have encountered an intermittent problem on Windows hosts which
+   presents as an 'Operation not permitted' error in the vagrant startup
+   output. If you encounter this error, perform a 'vagrant destroy' and
+   then run 'vagrant up' again.
+
+Downloading the Vagrant box file, booting the VM, and running through
+the bootstrap scripts will take several minutes.
+
+Once the 'vagrant up' command has finished executing, run:
+
+.. code-block:: console
+
+  % vagrant ssh
+
+By default, Vagrant sets up ssh keys so that users can log into the VM
+without setting up additional accounts or credentials. The logged in user,
+vagrant (uid 1000), also has permissions to execute sudo with no password
+required. Any number of `vagrant ssh` sessions can be established from the
+host.
+
+.. note::
+
+   Occasionally, the configuration files used to create the vagrant
+   environment change and this can result in vagrant asking for a password when
+   doing `vagrant up` or `vagrant ssh`. This is usually a sign that your
+   environment is out of date. If this happens, please follow the instructions
+   below to reset your environment. This is especially common when switching
+   between versions of Sawtooth Lake, eg. 0.7 -> 0.8.
+
+
+
+Resetting The Environment (Optional)
+------------------------------------
+
+If the VM needs to be reset for any reason, it can be returned to the default
+state by running the following commands from the sawtooth-core/tools directory
+on the host:
+
+.. code-block:: console
+
+  % vagrant destroy
+  % vagrant up
+
+.. warning::
+
+   vagrant destroy will delete all contents within the VM. However,
+   /vagrant and /project are shared with the host and will be preserved.
+
+
+Step Four: Build sawtooth-core
+==============================
+
+Most of the components of Sawtooth Lake depend partially on code that must
+first be built. This includes C++/swig code and generating protobuf classes for
+each language. Sawtooth Lake also includes SDKs for other languages, including
+Java and JavaScript, which require building. To build the python portion of
+Sawtooth Lake within vagrant, do:
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/build_python
+
+This will allow you to run and test Sawtooth Lake components manually. The
+automated tests for python and all other languages rely on docker to ensure
+reproducibility. To run the automated tests for python, first run the
+following:
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/build_all -l python
+
+This will create docker images for all the python components and run
+`build_python` inside a build container. You can then run the automated tests
+with:
+
+.. code-block:: console
+
+  $ /project/sawtooth-core/bin/run_tests -x java_sdk -x javascript_sdk
```

Comment:
Created new section. It should be expanded in another iteration (lots more to testing)